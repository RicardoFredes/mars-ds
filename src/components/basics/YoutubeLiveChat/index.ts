/**
 * WARNING!
 * Do not change this file,
 * because it is automatically generated
 * run "yarn reindex" to update
 */

import YoutubeLiveChat from "./youtube-live-chat.component";
export default YoutubeLiveChat;
export * from "./youtube-live-chat.types";
