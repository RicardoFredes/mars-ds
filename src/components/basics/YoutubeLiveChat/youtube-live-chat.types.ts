interface YouTubeLiveChatProps extends React.HTMLProps<HTMLIFrameElement> {
  domain?: string;
  height?: string;
  messageError?: string;
  messageMobile?: string;
  src?: string;
  videoId?: string;
  width?: string;
}

export type { YouTubeLiveChatProps };
