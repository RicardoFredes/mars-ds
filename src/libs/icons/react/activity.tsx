/**
 * WARNING!
 * Do NOT edit this file as it is automatically generated
 * run "yarn generate:icons" to update
 */

  
import type { IconLibProps } from "@/libs/icons/icons.types";

const IconActivity = ({ height = 24, width = 24 }: IconLibProps) => (
<svg
  xmlns="http://www.w3.org/2000/svg"
  width={width}
  height={height}
  viewBox="0 0 24 24"
  fill="none"
  stroke="currentColor"
  strokeWidth="2"
  strokeLinecap="round"
  strokeLinejoin="round"
>
  <polyline points="22 12 18 12 15 21 9 3 6 12 2 12" />
</svg>
);
  
export default IconActivity;