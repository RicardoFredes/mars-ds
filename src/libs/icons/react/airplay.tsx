/**
 * WARNING!
 * Do NOT edit this file as it is automatically generated
 * run "yarn generate:icons" to update
 */

  
import type { IconLibProps } from "@/libs/icons/icons.types";

const IconAirplay = ({ height = 24, width = 24 }: IconLibProps) => (
<svg
  xmlns="http://www.w3.org/2000/svg"
  width={width}
  height={height}
  viewBox="0 0 24 24"
  fill="none"
  stroke="currentColor"
  strokeWidth="2"
  strokeLinecap="round"
  stroke-linejoin="round"
>
  <path d="M5 17H4a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h16a2 2 0 0 1 2 2v10a2 2 0 0 1-2 2h-1" />
  <polygon points="12 15 17 21 7 21 12 15" />
</svg>
);
  
export default IconAirplay;