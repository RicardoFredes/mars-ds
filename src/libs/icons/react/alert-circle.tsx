/**
 * WARNING!
 * Do NOT edit this file as it is automatically generated
 * run "yarn generate:icons" to update
 */

  
import type { IconLibProps } from "@/libs/icons/icons.types";

const IconAlertCircle = ({ height = 24, width = 24 }: IconLibProps) => (
<svg
  xmlns="http://www.w3.org/2000/svg"
  width={width}
  height={height}
  viewBox="0 0 24 24"
  fill="none"
  stroke="currentColor"
  strokeWidth="2"
  strokeLinecap="round"
  strokeLinejoin="round"
>
  <circle cx="12" cy="12" r="10" />
  <line x1="12" y1="8" x2="12" y2="12" />
  <line x1="12" y1="16" x2="12.01" y2="16" />
</svg>
);
  
export default IconAlertCircle;