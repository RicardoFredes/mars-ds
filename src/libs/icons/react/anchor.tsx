/**
 * WARNING!
 * Do NOT edit this file as it is automatically generated
 * run "yarn generate:icons" to update
 */

  
import type { IconLibProps } from "@/libs/icons/icons.types";

const IconAnchor = ({ height = 24, width = 24 }: IconLibProps) => (
<svg
  xmlns="http://www.w3.org/2000/svg"
  width={width}
  height={height}
  viewBox="0 0 24 24"
  fill="none"
  stroke="currentColor"
  strokeWidth="2"
  strokeLinecap="round"
  strokeLinejoin="round"
>
  <circle cx="12" cy="5" r="3" />
  <line x1="12" y1="22" x2="12" y2="8" />
  <path d="M5 12H2a10 10 0 0 0 20 0h-3" />
</svg>
);
  
export default IconAnchor;