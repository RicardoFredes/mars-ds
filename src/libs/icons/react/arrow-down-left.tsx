/**
 * WARNING!
 * Do NOT edit this file as it is automatically generated
 * run "yarn generate:icons" to update
 */

  
import type { IconLibProps } from "@/libs/icons/icons.types";

const IconArrowDownLeft = ({ height = 24, width = 24 }: IconLibProps) => (
<svg
  xmlns="http://www.w3.org/2000/svg"
  width={width}
  height={height}
  viewBox="0 0 24 24"
  fill="none"
  stroke="currentColor"
  strokeWidth="2"
  strokeLinecap="round"
  strokeLinejoin="round"
>
  <line x1="17" y1="7" x2="7" y2="17" />
  <polyline points="17 17 7 17 7 7" />
</svg>
);
  
export default IconArrowDownLeft;