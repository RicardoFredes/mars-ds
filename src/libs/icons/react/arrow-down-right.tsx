/**
 * WARNING!
 * Do NOT edit this file as it is automatically generated
 * run "yarn generate:icons" to update
 */

  
import type { IconLibProps } from "@/libs/icons/icons.types";

const IconArrowDownRight = ({ height = 24, width = 24 }: IconLibProps) => (
<svg
  xmlns="http://www.w3.org/2000/svg"
  width={width}
  height={height}
  viewBox="0 0 24 24"
  fill="none"
  stroke="currentColor"
  stroke-width="2"
  stroke-linecap="round"
  stroke-linejoin="round"
>
  <line x1="7" y1="7" x2="17" y2="17" />
  <polyline points="17 7 17 17 7 17" />
</svg>
);
  
export default IconArrowDownRight;