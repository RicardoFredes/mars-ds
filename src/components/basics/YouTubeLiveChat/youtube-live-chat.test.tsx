import type { YouTubeLiveChatProps } from "./youtube-live-chat.types";

import { render } from "@testing-library/react";

import YoutubeLiveChat from "./youtube-live-chat.component";

const makeSut = (props?: YouTubeLiveChatProps) => render(<YoutubeLiveChat {...props} />);

describe("<YoutubeLiveChat>", () => {
  it("should render component", () => {
    const wrapper = makeSut();
    expect(wrapper).toBeTruthy();
  });

  it('should render component with "domain" and "videoId" props', () => {
    const wrapper = makeSut({
      domain: "localhost",
      videoId: "AmdW4DdvcQE",
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('should render component with "src" props', () => {
    const wrapper = makeSut({
      src: "https://www.youtube.com/live_chat?v=AmdW4DdvcQE&embed_domain=localhost",
    });
    expect(wrapper).toMatchSnapshot();
  });

  it("should render component with advanced props", () => {
    const wrapper = makeSut({
      domain: "localhost",
      height: "250px",
      videoId: "AmdW4DdvcQE",
      width: "250px",
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('should display specific text when property "videoId" or "src" is not input', () => {
    const wrapper = makeSut({
      messageError: "Não foi possível estabelecer conexão com o chat",
    });
    const element = wrapper.getByTestId("messageError");
    expect(element).toHaveTextContent("Não foi possível estabelecer conexão com o chat");
  });
});
