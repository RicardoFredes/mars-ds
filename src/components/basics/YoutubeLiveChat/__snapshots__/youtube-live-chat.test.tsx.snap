// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<YouTubeLiveChat> should render component with "domain" and "videoId" props 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="youtube-live-chat"
      >
        <iframe
          class="youtube-live-chat"
          data-testid="iframe"
          src="https://www.youtube.com/live_chat?v=AmdW4DdvcQE&embed_domain=localhost"
          style="width: 100%; height: 380px;"
        />
        <p
          class="text text--size-md"
          data-testid="messageMobile"
        >
          Não é possível exibir o chat em dispositivos móveis
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="youtube-live-chat"
    >
      <iframe
        class="youtube-live-chat"
        data-testid="iframe"
        src="https://www.youtube.com/live_chat?v=AmdW4DdvcQE&embed_domain=localhost"
        style="width: 100%; height: 380px;"
      />
      <p
        class="text text--size-md"
        data-testid="messageMobile"
      >
        Não é possível exibir o chat em dispositivos móveis
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<YouTubeLiveChat> should render component with "src" props 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="youtube-live-chat"
      >
        <iframe
          class="youtube-live-chat"
          data-testid="iframe"
          src="https://www.youtube.com/live_chat?v=AmdW4DdvcQE&embed_domain=localhost"
          style="width: 100%; height: 380px;"
        />
        <p
          class="text text--size-md"
          data-testid="messageMobile"
        >
          Não é possível exibir o chat em dispositivos móveis
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="youtube-live-chat"
    >
      <iframe
        class="youtube-live-chat"
        data-testid="iframe"
        src="https://www.youtube.com/live_chat?v=AmdW4DdvcQE&embed_domain=localhost"
        style="width: 100%; height: 380px;"
      />
      <p
        class="text text--size-md"
        data-testid="messageMobile"
      >
        Não é possível exibir o chat em dispositivos móveis
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<YouTubeLiveChat> should render component with advanced props 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="youtube-live-chat"
      >
        <iframe
          class="youtube-live-chat"
          data-testid="iframe"
          src="https://www.youtube.com/live_chat?v=AmdW4DdvcQE&embed_domain=localhost"
          style="width: 250px; height: 250px;"
        />
        <p
          class="text text--size-md"
          data-testid="messageMobile"
        >
          Não é possível exibir o chat em dispositivos móveis
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="youtube-live-chat"
    >
      <iframe
        class="youtube-live-chat"
        data-testid="iframe"
        src="https://www.youtube.com/live_chat?v=AmdW4DdvcQE&embed_domain=localhost"
        style="width: 250px; height: 250px;"
      />
      <p
        class="text text--size-md"
        data-testid="messageMobile"
      >
        Não é possível exibir o chat em dispositivos móveis
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
